### PR TITLE
added utility for pagination sort and filtering

### PR DESCRIPTION
**Pull request by PaulBredl**
,## Description of changes
Please provide a short description of the change or if the change is sufficiently described in the GitHub issue, provide a link to the issue.

  
## How has this been tested?
Please describe the test strategy you followed.

- [x] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [ ] manual, exploratory test

In case of manual test, please document the test well including a set of user instructions and prerequisites. Each including an action, it's result, and where appropriate a screenshot.
In case of automated tests, the test coverage should be reasonably high, especially on Service classes and other classes with complex logic.
    
## Checklist before requesting a review
- [x] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code fulfills all acceptance criteria
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [x] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: src/main/java/de/unistuttgart/iste/gits/common/util/PaginationUtil.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
+package de.unistuttgart.iste.gits.common.util;
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/common/util/PaginationUtil.java:1]: The three classes in this repo should be moved to a common code repository, as well as the tests

File: build.gradle [hunk 2]
```diff
@@ -14,7 +14,7 @@ graphqlCodegen {
 	// all config options:
 	// https://github.com/kobylynskyi/graphql-java-codegen/blob/main/docs/codegen-options.md
 	outputDir = new File("$buildDir/generated")
-	packageName = "de.unistuttgart.iste.gits.template.dto" // TODO: change to your package name
+	packageName = "de.unistuttgart.iste.gits.generated.dto"
```
Comment (PaulBredl) [inline build.gradle:17]: we cannot have different packages for each repo for the generated code, if we want to use the dtos in the common utility.
Here, the three utility classes added all require dtos, so we need to use the same package in all microservices.

**Comment by myluki2000** [comment]
Looks good now

**Review by myluki2000 — APPROVED** [review]
Looks good now
